api/src/main/java/org/openmrs/messagesource/impl/MessageSourceServiceImpl.java

```diff
@@ -96,7 +96,12 @@ public void setActiveMessageSource(MutableMessageSource activeMessageSource) {
 	 *      java.util.Locale)
 	 */
 	public String getMessage(MessageSourceResolvable resolvable, Locale locale) {
-		return activeMessageSource.getMessage(resolvable, locale);
+		if((resolvable.getCodes()[0]).equals((activeMessageSource.getMessage(resolvable, locale)))){
```

Comment:
I'm quite surprised that this is the correct fix. (I would expect that activeMessageSource.getMessageSource(MessageSourceResolvable, Locale) would be part of a Spring class that should already be behaving correctly to Spring's specifications....